RealmSwift/Impl/RealmCollectionImpl.swift

```diff
@@ -185,7 +185,15 @@ internal func with<A: Actor, Value: ThreadConfined, Return: Sendable>(
         let unchecked = Unchecked(wrappedValue: value)
         return try await actor.invoke { actor in
             if !Task.isCancelled {
+#if swift(>=5.10)
```

Comment:
I'm deducing this is only happening in 5.10, can we exclude future swift releases from this fix?

Reply:
It's fixed on swift main, but until there's a release with the fix we don't know exactly which versions to check.